### PR TITLE
Completed schemas for sources

### PR DESCRIPTION
**Pull request by attwad**
I can deal with the merge conflict if you want to submit your PR that moves stuff around first, go ahead.

### CODE REVIEW

**Comment by attwad** [comment]
[Design data ingestion config](https://trello.com/c/q2vkjYK3/70-design-data-ingestion-config)

File: verification/curator-service/src/model/source.ts [hunk 1]
```diff
@@ -15,17 +15,68 @@ const originSchema = new mongoose.Schema({
...
+        required: 'Enter a field extraction regexp',
+    },
+});
+
+interface Parsing {
+    fields: Array<Field>;
+}
+
+const parsingSchema = new mongoose.Schema({
+    fields: {
```
Comment (axmb) [inline verification/curator-service/src/model/source.ts:39]: nit: can this just be `fields: [fieldSchema],`?

File: verification/curator-service/src/model/source.ts [hunk 2]
```diff
@@ -15,17 +15,68 @@ const originSchema = new mongoose.Schema({
...
+    active: boolean;
+    scheduleExpression: string;
+    parsing: Parsing;
+}
+
+const automationSchema = new mongoose.Schema({
+    name: {
+        name: {
+            type: String,
+            required: 'Enter an automation rule name',
```
Comment (axmb) [inline verification/curator-service/src/model/source.ts:56]: Re: required: this is the "name of the image in docker registry" -- what if they're using the `parsing` field, instead?

File: verification/curator-service/src/model/source.ts [hunk 3]
```diff
@@ -15,17 +15,68 @@ const originSchema = new mongoose.Schema({
...
+        },
+        tag: String,
+        active: Boolean,
+        scheduleExpression: String,
```
Comment (axmb) [inline verification/curator-service/src/model/source.ts:60]: Something to consider down the line:

https://mongoosejs.com/docs/validation.html#custom-validators

File: verification/curator-service/src/model/source.ts (hunk 1 above)
Comment (attwad) [reply verification/curator-service/src/model/source.ts]: indeed, thanks

File: verification/curator-service/src/model/source.ts (hunk 3 above)
Comment (attwad) [reply verification/curator-service/src/model/source.ts]: Yeah I saw that, I'm waiting to see how we want to optionalize/make mandatory various manual vs automatic configs before adding more validation.
I left a TODO for a regexp validator like "cron(" or "rate("

**Pull request merged**